trio/tests/test_threads.py

```diff
@@ -471,6 +471,15 @@ def thread_fn():
     trio_time = await to_thread_run_sync(thread_fn)
     assert isinstance(trio_time, float)
 
+    # Test correct error when passed async function
+    async def async_fn():  # pragma: no cover
+        pass
+
+    with pytest.raises(TypeError) as excinfo:
+        from_thread_run_sync(async_fn)
+
+    assert "expected a sync function" in str(excinfo.value)
```

Comment:
It's nicer to write this as `with pytest.raises(TypeError, match="expected a sync function"):`